### PR TITLE
feat: add configuration for insert_file_link title

### PR DESCRIPTION
**Pull request by max397574**
fixes https://github.com/nvim-neorg/neorg-telescope/issues/59
supersedes https://github.com/nvim-neorg/neorg-telescope/pull/60
could you test this @f4z3r @benlubas

### CODE REVIEW

**Comment by f4z3r** [comment]
@max397574 ah nice `modules.get_module_config()` makes for a much nicer implementation indeed. I was not aware of that function sorry.

I tested it and it works great!

File: README.md [hunk 1]
```diff
@@ -170,6 +170,22 @@ You can install it through your favorite plugin manager:
...
   </details>
 
+# Configuration
+
+This module accepts the following configuration with the shown defaults:
+```lua
+{
+    ...
+    ["core.integrations.telescope"] = {
+        insert_file_link = {
```
Comment (f4z3r) [inline README.md:180]: This should be within a `config` block I think:

```lua
{
  ...
  ["core.integrations.telescope"] = {
    config = {
      insert_file_link = {
        show_title_preview = false,
      },
    },
  },
}
```
```

**Comment by max397574** [comment]
so this works for you and would close your pr and the issue right?
I also tried the config option and it felt faster even though the lag was nowhere as bad for me as described in the issue because I don't have such a big workspace

**Comment by f4z3r** [comment]
> so this works for you and would close your pr and the issue right? I also tried the config option and it felt faster even though the lag was nowhere as bad for me as described in the issue because I don't have such a big workspace

Yes, IMO it would close my PR and issue.

**Review by f4z3r — APPROVED** [review]
Do you purposefully use `table|nil` instead of `table?`? Otherwise it LGTM.

**Comment by max397574** [comment]
yes because it can return nil so it's needed to make lua-language-server happy
and I do check for nil

**Comment by f4z3r** [comment]
> yes because it can return nil so it's needed to make lua-language-server happy and I do check for nil

Hahahah I am aware of that. I meant why use the union style annotations rather than the question mark style for potentially nil values.

**Pull request merged**